packages/terra-functional-testing/tests/jest/services/wdio-docker-service.test.js

```diff
@@ -0,0 +1,302 @@
+      const service = new DockerService();
+
+      mockExec.mockImplementation(() => Promise.resolve({ stdout: 'Active' }));
+
+      jest.spyOn(service, 'awaitServiceRemoval').mockImplementationOnce(() => Promise.resolve());
+      jest.spyOn(service, 'awaitNetworkRemoval').mockImplementationOnce(() => Promise.resolve());
+
+      await service.removeStack();
+
+      expect(mockExec).toHaveBeenCalledTimes(2);
+      expect(mockExec).toHaveBeenCalledWith('docker stack rm wdio');
+    });
+
+    it('should wait for the network and service to be removed', async () => {
+      const service = new DockerService();
+
+      mockExec.mockImplementation(() => Promise.resolve({ stdout: 'Active' }));
+
+      jest.spyOn(service, 'awaitServiceRemoval').mockImplementationOnce(() => Promise.resolve());
+      jest.spyOn(service, 'awaitNetworkRemoval').mockImplementationOnce(() => Promise.resolve());
+
+      await service.removeStack();
+
+      expect(service.awaitServiceRemoval).toHaveBeenCalled();
+      expect(service.awaitNetworkRemoval).toHaveBeenCalled();
+    });
+  });
+
+  describe('pollCommand', () => {
+    it('should invoke the command and callback', async () => {
+      const service = new DockerService();
+
+      mockExec.mockImplementation(() => Promise.resolve('mock output'));
+
+      const mockCallback = jest.fn().mockImplementation(() => Promise.resolve());
+
+      await service.pollCommand('mock command', mockCallback);
+
+      expect(mockExec).toHaveBeenCalledWith('mock command');
+      expect(mockCallback).toHaveBeenCalledWith('mock output');
+    });
+
+    it('should timeout if the retry count is exceeded', async () => {
+      const service = new DockerService();
+
+      mockExec.mockImplementation(() => Promise.resolve());
+
+      const mockCallback = jest.fn().mockImplementation(() => Promise.reject());
+
+      try {
+        // Retry 3 times. Repeat every 10 milliseconds.
+        await service.pollCommand('mock command', mockCallback, 3, 10);
+      } catch (error) {
+        expect(mockCallback).toHaveBeenCalledTimes(3);
+      }
+    });
+  });
+
+  describe('awaitNetworkRemoval', () => {
+    it('should wait until the network has been removed', async () => {
+      const service = new DockerService();
+
+      jest.spyOn(service, 'pollCommand').mockImplementationOnce(() => Promise.resolve());
+
+      await service.awaitNetworkRemoval();
+
+      expect(service.pollCommand).toHaveBeenCalledWith('docker network ls | grep wdio || true', expect.any(Function));
+    });
+
+    it('should reject the command if the network status is present', async () => {
+      const service = new DockerService();
+
+      jest.spyOn(service, 'pollCommand').mockImplementationOnce(async (_command, callback) => {
+        await expect(callback({ stdout: 'Active Network' })).rejects.toBeUndefined();
+
+        return Promise.resolve();
+      });
+
+      await service.awaitNetworkRemoval();
+
```

Comment:
Is this is rejected, does the service onPrepare fail? Is a message logged or a stack trace provided to the user?

Reply:
If this is rejected or exceeds the retry count the timeout message will bubble and be logged to the console:

```
2020-06-02T21:22:49.364Z INFO @wdio/cli:launcher: Run onPrepare hook
[terra-functional-testing:wdio-docker-service] Deploying docker stack.
[terra-functional-testing:wdio-docker-service] Waiting for docker to become ready.
2020-06-02T21:22:58.681Z ERROR @wdio/cli:utils: A service failed in the 'onPrepare' hook
Error: [terra-functional-testing:wdio-docker-service] Timeout. Exceeded retry count.
    at Timeout.poll (/Users/se028266/workspaces/terra/terra-toolkit/packages/terra-functional-testing/lib/services/wdio-docker-service.js:86:18)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)

Continue...
```